pages/topics/new.tsx

```diff
@@ -19,15 +23,23 @@ type NewProps = {
 };
 
 function New({ edit, back, onSubmit }: NewProps) {
-  const [videoTracks, addVideoTrack] = useAddVideoTrackAtom();
+  const {
+    videoTracksProps,
+    resetVideoTrackProps,
+    addVideoTrack,
+    deleteVideoTrack,
+  } = useVideoTrackAtom();
+  useEffect(() => {
+    resetVideoTrackProps();
+  }, [resetVideoTrackProps]);
```

Comment:
これは resetVideoTrackProps の変更の都度実行されることを意味するのではないかと思うのですが問題ありませんか?
resetVideoTrackProps 自体のライフサイクルをよく追えていないので、誤っていたら申し訳ないです。

Reply:
やりたかったこととしては、コンポーネントのマウント時に一度だけ実行したい、ということであって、useEffectを使いたい強い意図があったわけではないです。どのように記述するとベストでしょう？

Reply:
```suggestion
  useEffect(() => {
    resetVideoTrackProps();
  }, []);
```

> これは resetVideoTrackProps の変更の都度実行されることを意味するのではないかと思うのですが問題ありませんか?

react-hooks/exhaustive-deps に引っかかると勝手に思っていましたが、このようにすれば都度実行しないことが明示的になるでしょうか

Reply:
useEffect によってマウント時に実行されることに関しては正しいです :ok_man: 
resetVideoTrackProps は addVideoTrack を呼ぶと変わらないんだろうかと気になったということです。
変化するならば、addVideoTrack() 呼ぶ → resetVideoTrackProps() となって永久に追加できないだろうなと。
不変ならば、そもそも deps 内に [resetVideoTrackProps] と加えずとも [] で済みそうかも。ESLintに怒られるだろうな。うむー。

Reply:
> resetVideoTracksProps 自体のライフサイクルをよく追えていない

私も確信をもってそうだという自信はないのですが、resetVideoTracksPropsを実行して値が変わるのはvideoTracksPropsであって、resetVideoTrackPropsは不変(だといいなあ)という認識です…
私の環境で試した分では、addVideoTrackの実行時にはvideoTracksPropsはクリアされずに正常に追加されるようです

Reply:
実際試した感じ問題無さそうだったので杞憂なので本件撤回します!